sympy/solvers/bivariate.py

```diff
@@ -156,10 +158,13 @@ def _lambert(eq, x):
             continue
         rhs = -c/b + (a/d)*l
 
-        solns = solve(X1 - u, x)
-        for i, tmp in enumerate(solns):
-            solns[i] = tmp.subs(u, rhs)
-            sol.append(solns[i])
+        solns = solveset_real(X1 - u, x)
+        if isinstance(solns, Intersection):
```

Comment:
Here the solns is like this `Intersection(Reals, {_rhs})` . I think it should give result like this `{_rhs}` if `_rhs` is dummy variable assuming to be real. What do you think @smichr ?

Reply:
Is the solution guaranteed to be in that form any time it is an `Intersection`? If not then you should check it is in that form before doing this.

Reply:
Actually No, because sometimes solns turn out to be like this `Intersection(Reals, {exp(_rhs)})` and in this case my PR fails. I need help to extract this part `exp(_rhs)` from Intersection.

But dont you think `Intersection(Reals, {_rhs})` should return `{_rhs}` if _rhs is dummy variable.

Reply:
```
In [17]: sol = FiniteSet(exp(x)).intersect(Reals)                                                                                              

In [18]: sol                                                                                                                                   
Out[18]: 
    ⎧ x⎫
ℝ ∩ ⎨ℯ ⎬
    ⎩  ⎭

In [19]: repr(sol)                                                                                                                             
Out[19]: 'Intersection(Reals, {exp(x)})'

In [20]: sol.func                                                                                                                              
Out[20]: sympy.sets.sets.Intersection

In [21]: sol.args                                                                                                                              
Out[21]: 
⎛   ⎧ x⎫⎞
⎜ℝ, ⎨ℯ ⎬⎟
⎝   ⎩  ⎭⎠

In [22]: sol.args[0] is Reals                                                                                                                  
Out[22]: True

In [23]: len(sol.args[1])                                                                                                                      
Out[23]: 1

In [24]: list(sol.args[1])[0]                                                                                                                  
Out[24]: 
 x
ℯ 
```